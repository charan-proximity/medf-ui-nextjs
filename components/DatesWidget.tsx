import React from "react";
import { SlotDateType } from "../types";

interface IDatesWidgetProps {
  slotBookingDateHeaders: SlotDateType[];
  onDateChange: (date: SlotDateType) => void;
  selectedDate: SlotDateType;
}

function DatesWidget({
  slotBookingDateHeaders,
  onDateChange,
  selectedDate,
}: IDatesWidgetProps) {
  const selectedDayClass = "text-primary-pink  p-0.5";
  const selectedDateClass = "bg-primary-pink text-white p-2 rounded-full ";
  return (
    <div className=" h-20  overflow-hidden  shadow-xl ">
      <ul className="flex p-2 flex-row justify-between mx-10 items-center h-full ">
        {slotBookingDateHeaders.map((head: SlotDateType) => (
          <li
            className=""
            key={head.fullDate}
            onClick={() => onDateChange(head)}
          >
            <div className="text-xs flex flex-col ">
              <div
                className={` ${
                  selectedDate.dayName == head.dayName ? selectedDayClass : ""
                }`}
              >
                {head.dayName}
              </div>
              <div
                className={` ${
                  selectedDate.day == head.day
                    ? selectedDateClass
                    : "mt-2.5 pt-0.5"
                }`}
              >
                <p className=""> {head.day}</p>
              </div>
            </div>
          </li>
        ))}
      </ul>
    </div>
  );
}

export default DatesWidget;
