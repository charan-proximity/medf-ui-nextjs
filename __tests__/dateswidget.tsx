import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import DatesWidget from "../components/DatesWidget";

describe(" Date Widget tests", () => {
  const slotBookingDateHeaders = [
    { fullDate: "Tue, 28 June 2022", dayName: "Tue", day: "28" },
    { fullDate: "Wed, 29 June 2022", dayName: "Wed", day: "29" },
    { fullDate: "Thu, 30 June 2022", dayName: "Thu", day: "30" },
    { fullDate: "Fri, 1 July 2022", dayName: "Fri", day: "01" },
    { fullDate: "Sat, 2 July 2022", dayName: "Sat", day: "02" },
    { fullDate: "Sun, 3 July 2022", dayName: "Sun", day: "03" },
    { fullDate: "Mon, 4 July 2022", dayName: "Mon", day: "04" },
  ];
  const selectedDate = slotBookingDateHeaders[0];
  const onDateChange = jest.fn();

  test("should be able to render the dates ", async () => {
    const { debug } = render(
      <DatesWidget
        slotBookingDateHeaders={slotBookingDateHeaders}
        selectedDate={selectedDate}
        onDateChange={onDateChange}
      />
    );
    const listtestems = screen.getAllByRole("listitem");
    expect(listtestems).toHaveLength(slotBookingDateHeaders.length);
    expect(listtestems).toMatchSnapshot();
  });
});
