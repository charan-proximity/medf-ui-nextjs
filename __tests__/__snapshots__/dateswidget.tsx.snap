// Jest Snapshot v1, https://goo.gl/fbAQLP



exports[` Date Widget tests should be able to render the dates  1`] = `
Array [
  <li
    class=""
  >
    <div
      class="text-xs flex flex-col "
    >
      <div
        class=" text-primary-pink  p-0.5"
      >
        Tue
      </div>
      <div
        class=" bg-primary-pink text-white p-2 rounded-full "
      >
        <p
          class=""
        >
           
          28
        </p>
      </div>
    </div>
  </li>,
  <li
    class=""
  >
    <div
      class="text-xs flex flex-col "
    >
      <div
        class=" "
      >
        Wed
      </div>
      <div
        class=" mt-2.5 pt-0.5"
      >
        <p
          class=""
        >
           
          29
        </p>
      </div>
    </div>
  </li>,
  <li
    class=""
  >
    <div
      class="text-xs flex flex-col "
    >
      <div
        class=" "
      >
        Thu
      </div>
      <div
        class=" mt-2.5 pt-0.5"
      >
        <p
          class=""
        >
           
          30
        </p>
      </div>
    </div>
  </li>,
  <li
    class=""
  >
    <div
      class="text-xs flex flex-col "
    >
      <div
        class=" "
      >
        Fri
      </div>
      <div
        class=" mt-2.5 pt-0.5"
      >
        <p
          class=""
        >
           
          01
        </p>
      </div>
    </div>
  </li>,
  <li
    class=""
  >
    <div
      class="text-xs flex flex-col "
    >
      <div
        class=" "
      >
        Sat
      </div>
      <div
        class=" mt-2.5 pt-0.5"
      >
        <p
          class=""
        >
           
          02
        </p>
      </div>
    </div>
  </li>,
  <li
    class=""
  >
    <div
      class="text-xs flex flex-col "
    >
      <div
        class=" "
      >
        Sun
      </div>
      <div
        class=" mt-2.5 pt-0.5"
      >
        <p
          class=""
        >
           
          03
        </p>
      </div>
    </div>
  </li>,
  <li
    class=""
  >
    <div
      class="text-xs flex flex-col "
    >
      <div
        class=" "
      >
        Mon
      </div>
      <div
        class=" mt-2.5 pt-0.5"
      >
        <p
          class=""
        >
           
          04
        </p>
      </div>
    </div>
  </li>,
]
`;
